### PR TITLE
Allow dataprovider builders to return GentyArgs.

### PR DESCRIPTION
**Pull request by Jeff-Meadows**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @Jeff-Meadows has signed the CLA. Thanks for the pull request!

File: HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,12 @@
 Release History
 ---------------
 
+1.2.2 (2015-04-09)
```
Comment (potrebic) [inline HISTORY.rst]: This date is going to be wrong. Is a specific 'date' needed?

File: test/test_genty.py [hunk 2]
```diff
@@ -493,3 +493,17 @@ class SomeClass(object):
...
+            def builder(self):
+                return genty_args(42, named='named_arg')
+
+            @genty_dataprovider(builder)
+            def test_method(self, number, default=None, named=None):
+                return number, default, named
+
+        instance = TestClass()
+        # pylint:disable=no-member
+        self.assertItemsEqual(instance.test_method_builder(), (42, None, 'named_arg'))
```
Comment (potrebic) [inline test/test_genty.py]: I believe that in most asserts we put the 'expected' data first, followed by the actual data. So I'd swap the order of the 2 params.

**Comment by potrebic** [comment]
couple minor comments. But looks good. +2

File: HISTORY.rst (hunk 1 above)
Comment (Jeff-Meadows) [inline HISTORY.rst]: Yes. I'll change it before pushing.

**Pull request merged**